### ISSUE TITLE
Moved Snipe-IT to https:// and now API doesn't work

### ISSUE THREAD

**Issue by Christopheric1** [snazy2000/SnipeitPS#260]
I have just moved from Centos7-http to Debian11-https and I am having problems getting the "Connect-SnipeITPS" to work. I have generated a new API key multiple times just to make sure I did it correctly. 

If I try to use "Invoke-WebRequest", this is what I get:
```
PS C:\Users\User> Invoke-WebRequest -Uri 'https://snipe-it1.fqdn/' -Verbose 
VERBOSE: GET https://snipe-it1.fqdn/ with 0-byte payload
Invoke-WebRequest : The request was aborted: Could not create SSL/TLS secure channel.
At line:1 char:1
+ Invoke-WebRequest -Uri 'https://snipe-it1.fqdn/' -Verbose
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```

I have tried to add the following above my command:
```
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
```

This made no difference. I am at a loss as to what to try to get this working. 

Thanks for your help!

**Comment by PetriAsi** [comment]
this is probably caused by custom or otherwise untrusted server certificate. Try to add you server ssl certificate or issued ca certificate to client machine as trusted. 

To test that your certificate is  trusted  just run  Invoke-WebRequest -Uri 'https://snipe-it1.fqdn/'
After you don't get any errors also snipeitps should work.

**Comment by Christopheric1** [comment]
Thanks for replying to the issue. I can verify that I have the certificate in "Trusted Root Certification Authority" and that is the same Cert that is on the Snipe-IT server. I don't get any certificate errors when I go to my https://snipe-it1.fqdn/ site. 

Is there anything else I can try?

**Comment by PetriAsi** [comment]
Sorry I , missed that you already tried with plain powershell and invoke-webrequest . So this cleary something between you server and powershell. Powershell uses .net ssl settings as you already tried force tls1.2, but this time this something else.

try to run https://www.ssllabs.com/ssltest/ against your server, after test you can see detailed results, what cipher suites, ssl protocols etc are supported by your server. Then you have to "just" make it compatible with your powershell environment.

**Comment by Christopheric1** [comment]
I was unable to do this as it is internal only. What I ended up checking was the .conf file in the SSL virtual host configuration file. In my case, it had the following in /etc/apache2/sites-enabled/000-snipeit.conf:

```
 SSLProtocol             all -SSLv3 -TLSv1 -TLSv1.1 -TLSv1.2
```
I changed it to allow TLSv1.2 by putting a "+" in front of it
```
 SSLProtocol             all -SSLv3 -TLSv1 -TLSv1.1 +TLSv1.2
```
Then, I added this line to the top of my script and it is working now
```
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
```
Thanks for at least getting me on the right track!
